crates/chain-state/src/memory_overlay.rs

```diff
@@ -153,13 +164,13 @@ impl StateRootProvider for MemoryOverlayStateProvider {
 impl StateProofProvider for MemoryOverlayStateProvider {
     fn hashed_proof(
         &self,
-        hashed_state: HashedPostState,
+        state: HashedPostState,
         address: Address,
         slots: &[B256],
     ) -> ProviderResult<AccountProof> {
-        let mut state = self.hashed_post_state.clone();
-        state.extend(hashed_state);
-        self.historical.hashed_proof(state, address, slots)
+        let MemoryOverlayTrieState { mut hashed_state, .. } = self.trie_state();
```

Comment:
atm we are reduntantly cloning in-memory nodes here, but this is fine since they will have to be used in computation anyway (as mentioned in todo above)